File: pinyougou-manager-web/src/main/webapp/js/service/specificationOptionService.js
// 服务层
pyg.service('specificationOptionService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('../specificationOption/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('../specificationOption/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('../specificationOption/findOne.do?id=' + id);
    };

    // 增加
    this.add = function (entity) {
        return $http.post('../specificationOption/add.do', entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('../specificationOption/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../specificationOption/batcheDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../specificationOption/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});

File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/SpecificationOptionService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbSpecificationOption;
import vo.PageResult;


/**
 * 服务层接口
 */
public interface SpecificationOptionService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbSpecificationOption> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbSpecificationOption specificationOption, int pageNum, int pageSize);


    /**
     * 添加
     */
    void add(TbSpecificationOption specificationOption);


    /**
     * 修改
     */
    void update(TbSpecificationOption specificationOption);


    /**
     * 根据 ID 获取实体
     *
     * @param id
     * @return
     */
    TbSpecificationOption findOne(long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);

}
